libzkbob-rs-wasm/src/client/mod.rs

```diff
@@ -316,7 +326,97 @@ impl UserAccount {
             });
         });
 
-        Ok(())
+        ()
+    }
+
+    
+    #[wasm_bindgen(js_name = "updateStateColdStorage")]
+    pub fn update_state_cold_storage(
+        &mut self,
+        bulks: Vec<js_sys::Uint8Array>,
+        from_index: Option<u64>,    // inclusively
+        to_index: Option<u64>,      // exclusively
+    ) -> Result<ParseTxsColdStorageResult, JsValue> {
+        const MAX_SUPPORTED_BULK_VERSION: u8 = 1;
+
+        let mut total_txs_cnt: usize = 0;
+        let bulks_obj: Result<Vec<BulkData>, JsValue> = bulks.into_iter().map(|array| {
+            let bulk_data = array.to_vec();
+            let bulk: BulkData = match bincode::deserialize(&bulk_data) {
+                Ok(res) => res,
+                Err(e) => return Err(js_err!(&format!("Cannot parse bulk data: {}", e))),
+            };
+
+            if bulk.bulk_version > MAX_SUPPORTED_BULK_VERSION {
+                return Err(js_err!(&format!("Incorrect bluk vesion {}, supported {}", bulk.bulk_version, MAX_SUPPORTED_BULK_VERSION)))
+            }
+
+            Ok(bulk)
+        })
+        .collect();
+
+        if let Err(e) = bulks_obj {
+            return Err(e);
+        }
+
+
+        let mut single_result: ParseResult = bulks_obj
+            .unwrap()
+            .into_iter()
+            .map(|bulk| -> Vec<ParseResult> {
+                let eta = &self.inner.borrow().keys.eta;
+                let params = &self.inner.borrow().params;
+                let range = from_index.unwrap_or(0)..to_index.unwrap_or(u64::MAX);
+                let bulk_results: Vec<ParseResult> = bulk.txs
+                    .into_par_iter()
```

Comment:
`into_par_iter` doesn't work in singlecore version of wasm. It should be replaced with `helpers::vec_into_iter` when we'll be merging this pr with https://github.com/zkBob/libzkbob-rs/pull/32.

Reply:
Okay, do not forget it is the main aim :)